test/session.test.ts

```diff
@@ -732,6 +732,55 @@ describe("Encoding entities", () => {
       expect(data[2].status.name).toEqual("Done");
     });
 
+    it("Should support denormalize feature toggle", () => {
+      const payload = [
+        {
+          id: 1,
+          __entity_type__: "Task",
+          name: "foo",
+          status: {
+            __entity_type__: "Status",
+            id: 2,
+            name: "In progress",
+          },
+        },
+        {
+          id: 2,
+          __entity_type__: "Task",
+          name: "foo",
+          status: {
+            __entity_type__: "Status",
+            id: 1,
+            name: "Done",
+          },
+        },
+        {
+          id: 3,
+          __entity_type__: "Task",
+          status: {
+            __entity_type__: "Status",
+            id: 1,
+          },
+        },
+      ];
+      // @ts-ignore - Otherwise internal method used for testing purposes
+      const normalizedData = session.decode(
```

Comment:
Feels cleaner to do this test on the public facing level of the API, and don't think that would make the test any more complicated.